src/ripple/nodestore/impl/DatabaseShardImp.h

```diff
@@ -242,6 +250,16 @@ class DatabaseShardImp : public DatabaseShard
     // File name used to mark shards being imported from node store
     static constexpr auto importMarker_ = "import";
 
+    // latestShardIndex_ and secondLatestShardIndex hold the indexes
+    // of the shards most recently confirmed by the network. These
+    // values are not updated in real time and are modified only
+    // when adding shards to the database, in order to determine where
+    // pending shards will be stored on the filesystem. A value of
+    // boost::none indicates that the corresponding shard is not held
+    // by the database.
+    boost::optional<std::uint32_t> latestShardIndex_;
+    boost::optional<std::uint32_t> secondLatestShardIndex_;
```

Comment:
Instead of `latest` and `secondLatest`, what do you think about `last` and `penultimate`? I think it's fine and clear the way it is, but "penultimate" means "next to last", and it's a little shorter, sooooooooo......

Reply:
Think I'll opt to keep the names as they are.

Reply:
> Think I'll opt to keep the names as they are.

I'm fine with that.